FTBFS: gcc/boost error in universe/ShipDesign.cpp

Compiling with gcc 6.3.0, boost 1.62.0.1 on Debian Linux

universe/ShipDesign.cpp:1608:23: error: invalid initialization of non-const reference of type ‘const char*&’ from an rvalue of type ‘const char*’
        DebugLogger() << "Populating pre-defined ships with "
                      << (are_monsters ? "monster":"ship") << " designs.";
It appears boost with gcc does not like alternates.
Commenting out this statement allows compilation to complete.

--- a/universe/ShipDesign.cpp
+++ b/universe/ShipDesign.cpp
@@ -1605,7 +1605,7 @@ namespace {
             return;
 
         DebugLogger() << "Populating pre-defined ships with "
-                      << (are_monsters ? "monster":"ship") << " designs.";
+                      << std::string(are_monsters ? "monster" : "ship") << " designs.";
 
         FillDesignsOrderingAndNameTables(
             *parsed, designs, ordering, name_to_uuid);

